### ISSUE TITLE
False positive E1133 Non-iterable value used in an iterating context (not-an-iterable)

### ISSUE THREAD

**Issue by dllu** [PyCQA/pylint#6044]
### Bug description

```python
class Wat:
    asdf = None
    @staticmethod
    def get_asdf():
        Wat.asdf = [1, 2, 3]
        return Wat.asdf

    def __init__(self):
        zxcv = [x + 1 for x in Wat.get_asdf()]
        self.zxcv = zxcv

wat = Wat()
print(wat.zxcv)
```

### Command used

```shell
pylint wtf.py
```

### Pylint output

```shell
wtf.py:9:31: E1133: Non-iterable value Wat.get_asdf() is used in an iterating context (not-an-iterable)
```

### Expected behavior

There shouldn't be a pylint error since `get_asdf()` always sets `Wat.asdf` to `[1, 2, 3]` before returning it.

### Pylint version

```shell
pylint 2.11.0
astroid 2.8.6
Python 3.8.5 | packaged by conda-forge | (default, Sep 24 2020, 16:55:52)
[GCC 7.5.0]
```

### OS / Environment

Ubuntu 18.04.3 LTS x86_64
